File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/models/ApiResponse.java
package com.timsmithsoftware.integration_tests.models;
import org.json.JSONObject;

public class ApiResponse {
	private int code;
	private JSONObject body;
	
	public ApiResponse(int code, JSONObject body) {
		this.code = code;
		this.body = body;
	}
	

    @Override
    public boolean equals(Object obj) {
        if (obj != null) {
            if (obj.getClass() == this.getClass()) {
            	final ApiResponse other = (ApiResponse) obj;
            	String otherBody = other.body.toString();
            	String thisBody = this.body.toString();
            	Boolean stringBodyEquality = otherBody.equals(thisBody);
            	if(this.code == other.code && stringBodyEquality) {
            		return true;
            	}
            }
        }
        return false;
    }
}

File: testing/backend/integration/integration-tests/src/main/java/com/timsmithsoftware/integration_tests/models/ApiResponse.kt
package com.timsmithsoftware.integration_tests.models

class ApiResponse(private val code: Int, private val body: Any?) {
    override fun equals(other: Any?): Boolean {
        if (other != null) {
            if (other.javaClass == this.javaClass) {
                val otherResponse = other as ApiResponse
                val otherBody = otherResponse.body.toString()
                val thisBody = body.toString()
                val stringBodyEquality = otherBody == thisBody
                if (this.code == otherResponse.code && stringBodyEquality) {
                    return true
                }
            }
        }
        return false
    }

    override fun toString(): String {
        return "Code: $code, Body: $body"
    }

    override fun hashCode(): Int {
        var result = code
        result = 31 * result + (body?.hashCode() ?: 0)
        return result
    }
}